domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -3463,20 +3496,52 @@ public void getCQLWrapperWithFacets(TestContext context) throws IOException, Fie
             context.assertEquals(1, resultInfo.getTotalRecords());
             List<Object> objs = handler.result().getResults();
             ObjectMapper mapper = new ObjectMapper();
-            context.assertEquals("{\"count\":1}", mapper.writeValueAsString(objs.get(0)));
-            context.assertEquals("{\"facetValues\":[{\"count\":1,\"value\":\"First edition\"}],\"type\":\"edition\"}",
-              mapper.writeValueAsString(objs.get(1)));
-            context.assertEquals("70fb4e66-cdf1-11e8-a8d5-f2801f1b9fd1",
-              new JsonObject(mapper.writeValueAsString(objs.get(2))).getString("id"));
             List<Facet> retFacets = resultInfo.getFacets();
+
             context.assertEquals(1, retFacets.size());
+            context.assertEquals("edition", retFacets.get(0).getType());
+            context.assertEquals(1, retFacets.get(0).getFacetValues().get(0).getCount());
+            context.assertEquals("First edition", retFacets.get(0).getFacetValues().get(0).getValue().toString());
+            /* because we're dealing using Object.class, the count object will be part of results */
+            context.assertEquals(2, objs.size()); // one count + one record
+            context.assertEquals("{\"count\":1}", mapper.writeValueAsString(objs.get(0)));
+            context.assertEquals("70fb4e66-cdf1-11e8-a8d5-f2801f1b9fd1",
+              new JsonObject(mapper.writeValueAsString(objs.get(1))).getString("id"));
           } catch (Exception ex) {
             context.fail(ex);
           }
           async.complete();
         });
       async.awaitSuccess();
     }
+
+    {
+      CQLWrapper cqlWrapper = new CQLWrapper(cql2pgJson, "order_format==Other");
+      Async async = context.async();
+      postgresClient.get(MOCK_POLINES_TABLE, Poline.class, "*",
+          cqlWrapper, true, true, facets, distinctOn, handler -> {
+            context.assertTrue(handler.succeeded());
+
+            try {
+              ResultInfo resultInfo = handler.result().getResultInfo();
+              context.assertEquals(1, resultInfo.getTotalRecords());
+              List<Poline> objs = handler.result().getResults();
+              List<Facet> retFacets = resultInfo.getFacets();
+
+              context.assertEquals(1, retFacets.size());
+              context.assertEquals("edition", retFacets.get(0).getType());
+              context.assertEquals(1, retFacets.get(0).getFacetValues().get(0).getCount());
+              context.assertEquals("First edition", retFacets.get(0).getFacetValues().get(0).getValue().toString());
+              context.assertEquals(1, objs.size());
+              context.assertEquals("70fb4e66-cdf1-11e8-a8d5-f2801f1b9fd1", objs.get(0).getId());
+            } catch (Exception ex) {
+              context.fail(ex);
+            }
```

Comment:
The lines

            context.assertTrue(handler.succeeded());

            try {

            } catch (Exception ex) {
              context.fail(ex);
            }

can be removed when replacing `handler -> {` by `context.asyncAssertSuccess(result -> {`.

Reply:
OK